modules/core/src/main/scala/io/chrisdavenport/ember/codec/Shared.scala

```diff
@@ -19,4 +22,23 @@ object Shared {
         ByteVector(bs.values, bs.offset, bs.size)
     }
   }
+
+  /** evaluates address from the host port and scheme, if this is a custom scheme we will default to port 8080**/
+  def addressForRequest[F[_]: Sync](req: Request[F]): F[InetSocketAddress] = 
+    for {
+      scheme <- req.uri.scheme.toRight(new Throwable("Missing Scheme")).liftTo[F]
+      host <- req.uri.host.toRight(new Throwable("Missing Host")).liftTo[F]
+      socketAddress <- addressForComponents(scheme, host, req.uri.port)
+    } yield socketAddress
+    
+  def addressForComponents[F[_] : Sync](scheme: Uri.Scheme, host: Uri.Host, port: Option[Int]): F[InetSocketAddress] = Sync[F].delay {
+    val finalPort = port.getOrElse {
+      scheme match {
+        case Uri.Scheme.https => 443
+        case Uri.Scheme.http => 80
+        case _ => 8080
+      }
+    }
+    new InetSocketAddress(host.value, finalPort)
```

Comment:
I can never remember what triggers a reverse DNS lookup, but I _think_ you want `createUnresolved` here.